src/components/Card.js

```diff
@@ -1,40 +1,71 @@
 export default class Card {
-  constructor({ title, link }, cardSelector, handleImageClick) {
-    this._title = title;
+  constructor(
+    { isLiked, _id, name, link },
+    cardSelector,
+    handleImageClick,
+    handleDeleteClick,
+    handleCardLike
+  ) {
+    this._isLiked = isLiked;
+    this._id = _id;
+    this._name = name;
     this._link = link;
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
+    this._handleDeleteClick = handleDeleteClick;
+    this._handleCardLike = handleCardLike;
   }
 
   _setEventListeners() {
     // card like button
     const heart = this._cardElement.querySelector(".heart-button");
     heart.addEventListener("click", () => {
       this._handleHeartButton();
```

Comment:
`      this._handleHeartButton();
`
You need to **toggle likes** only in block `then` after a successful response from the server because if there is an error, the user will not know about that and will see the toggled heart color.